ivy/functional/backends/numpy/experimental/manipulation.py

```diff
@@ -483,3 +487,25 @@ def column_stack(
     arrays: Sequence[np.ndarray], /, *, out: Optional[np.ndarray] = None
 ) -> np.ndarray:
     return np.column_stack(arrays)
+
+
+@with_supported_dtypes(
+    {"1.25.2 and below": ("float32", "float64", "int32", "int64")}, backend_version
+)
+def put_along_axis(
+    arr: np.ndarray,
+    indices: np.ndarray,
+    values: np.ndarray,
+    axis: int,
+    /,
+    *,
+    mode: str = None,
+    out: Optional[np.ndarray] = None,
+):
+    np.put_along_axis(arr, indices, values, axis)
+    return ivy.inplace_update(out, arr) if ivy.exists(out) else arr
+
+
+put_along_axis.partial_mixed_handler = lambda *args, mode=None, **kwargs: mode in [
+    "assign",
+]
```

Comment:
Oh and the `mode` typehints should be the same as in the main implementation. I updated it in torch but it also needs to change in numpy and paddle.

Reply:
Numpy only supports `replace` and Paddle doesn't support `mean`, `max`, or `mean`. The function is a partial mixed function, so the function jumps directly to the implemented `put_along_axis` backends without running `scatter_nd` in the compositional implementation